integration_tests/src/lib.rs

```diff
@@ -0,0 +1,184 @@
+            });
+        }
+
+        if let Some(log_data) = logs {
+            // TODO
+            insta::assert_yaml_snapshot!(format!("{}_logs", prefix), log_data);
+        }
+    }
+
+    async fn get_telemetry_from_spin_app(
+        path: &str,
+    ) -> Result<(Vec<ExportedSpan>, Vec<ExportedMetric>, Vec<ExportedLog>), Error> {
+        // Start the collector.
+        let (mut collector, collector_endpoint) = start_collector().await;
+
+        // Build, instantiate, and invoke the Spin app.
+        let _guard = ProcessGuard("spin");
+        build_spin_app(path)?;
+        instantiate_spin_app(path, &collector_endpoint)?;
+        invoke_spin_app().await?;
+
+        // Retrieve telemetry data.
+        let timeout = Duration::from_secs(10);
+        let collector_min = 6;
+        let spans = collector.exported_spans(collector_min, timeout).await;
+        let metrics = collector.exported_metrics(collector_min, timeout).await;
+        let logs = collector.exported_logs(collector_min, timeout).await;
+
+        Ok((spans, metrics, logs))
+    }
+
+    /// Starts a fake collector server.
+    async fn start_collector() -> (FakeCollectorServer, String) {
+        let collector = FakeCollectorServer::start()
+            .await
+            .expect("fake collector should start");
+        let endpoint = collector.endpoint().clone();
+        (collector, endpoint)
+    }
+
+    /// Sends an HTTP request to a Spin app.
+    async fn invoke_spin_app() -> Result<(), Error> {
+        let start = std::time::Instant::now();
+        loop {
+            match reqwest::get("http://localhost:3000").await {
+                Ok(_) => return Ok(()),
+                Err(e) => {
+                    if start.elapsed() > Duration::from_secs(5) {
+                        return Err(anyhow!("Unable to reach the Spin app: {e}"));
+                    }
+                }
+            }
+        }
+    }
+
+    /// Compiles a Spin app.
+    fn build_spin_app(path: &str) -> Result<(), Error> {
+        let spin_build_output = Command::new("spin")
+            .args(["build", "-f", path])
+            .output()
+            .expect("Failed to execute 'spin build'");
+        if !spin_build_output.status.success() {
+            return Err(anyhow!(
+                "{} -> build: {}",
+                path,
+                String::from_utf8_lossy(&spin_build_output.stderr)
+            ));
+        }
+        Ok(())
+    }
+
+    /// Instantiates a Spin app.
+    fn instantiate_spin_app(path: &str, collector_endpoint: &str) -> Result<(), Error> {
+        Command::new("spin")
+            .env("OTEL_EXPORTER_OTLP_ENDPOINT", collector_endpoint)
+            .env("OTEL_EXPORTER_OTLP_PROTOCOL", "grpc")
+            .args(["up", "-f", path])
+            .spawn()
+            .map(|_| ())
+            .map_err(|e| anyhow!(e))
+    }
+
+    /// Once dropped, this will kill all processes with names that contain the substring.
```

Comment:
If you put the command run in here you wouldn't have to kill everything. You could just kill the PID.